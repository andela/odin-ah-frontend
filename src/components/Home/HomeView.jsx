import React from 'react';
import PropTypes from 'prop-types';
import Header from '../Header/HeaderContainer';
import './Home.scss';
import ArticleGridView from '../ArticleList/ArticleGridView';

const HomeView = props => (
    <div className="App">
      <Header isAuthenticated={false} />
      <div>
        <section className="section">
          <div className="container">
            <div className="columns is-centered">
              <div className="column has-text-centered title section-title">Top Articles</div>
            </div>
            <ArticleGridView articles={props.articles} />
          </div>
        </section>
        <footer className="footer">
          <div className="content has-text-centered">
            <p>&copy; Authors Haven 2018</p>
          </div>
        </footer>
      </div>
    </div>
);

HomeView.propTypes = {
  articles: PropTypes.array
};

export default HomeView;
