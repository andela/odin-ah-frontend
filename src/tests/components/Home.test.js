import React from 'react';
import { shallow } from 'enzyme';
import HomeView from '../../components/Home/HomeView';

it('renders HomeView without crashing', () => {
  const wrapper = shallow(<HomeView />);
  expect(wrapper).toMatchSnapshot();
});
