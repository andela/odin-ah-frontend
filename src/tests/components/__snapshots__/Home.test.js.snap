// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomeView without crashing 1`] = `
<div
  className="App"
>
  <Connect(HeaderContainer)
    isAuthenticated={false}
  />
  <div>
    <section
      className="section"
    >
      <div
        className="container"
      >
        <div
          className="columns is-centered"
        >
          <div
            className="column has-text-centered title section-title"
          >
            Top Articles
          </div>
        </div>
        <ArticleGridView />
      </div>
    </section>
    <footer
      className="footer"
    >
      <div
        className="content has-text-centered"
      >
        <p>
          Â© Authors Haven 2018
        </p>
      </div>
    </footer>
  </div>
</div>
`;
